### PR TITLE
fix: use safe reader in container-based build tool

### PR DESCRIPTION
**Pull request by asraa**
Part of https://github.com/slsa-framework/slsa-github-generator/issues/1893

### CODE REVIEW

File: internal/builders/docker/pkg/config.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func NewDockerBuildConfig(io *InputOptions) (*DockerBuildConfig, error) {
 		return nil, err
 	}
 
-	if err = validatePath(io.BuildConfigPath); err != nil {
+	if err = utils.PathIsUnderCurrentDirectory(io.BuildConfigPath); err != nil {
```
Comment (laurentsimon) [inline internal/builders/docker/pkg/config.go:89]: this is an early validation, and the main builder will use a safeRead anyway, correct?

File: internal/builders/docker/pkg/config.go [hunk 2]
```diff
@@ -168,14 +160,18 @@ func (dbc *DockerBuildConfig) LoadBuildConfigFromFile() (*BuildConfig, error) {
 // not exposed. The corresponding method LoadBuildConfigFromFile must be called
 // on an instance of DockerBuildConfig which has a validated BuildConfigPath.
 func loadBuildConfigFromFile(path string) (*BuildConfig, error) {
-	tomlTree, err := toml.LoadFile(path)
+	tomlFile, err := utils.SafeReadFile(path)
```
Comment (laurentsimon) [inline internal/builders/docker/pkg/config.go:163]: nit: rename to `tomFileBytes`

**Review by laurentsimon — APPROVED** [review]

File: internal/builders/docker/pkg/builder_test.go [hunk 3]
```diff
@@ -162,12 +162,23 @@ func Test_GitClient_invalidSourceRef(t *testing.T) {
...
 	out := t.TempDir()
+
 	wd, err := os.Getwd()
 	if err != nil {
 		t.Fatal(err)
 	}
+	// Execute this function from docker/ instead of pkg/ so that the testdata
+	// folder is in the current dir.
+	t.Cleanup(func() {
+		if err := os.Chdir(wd); err != nil {
```
[the remark this answers is not in the document]
Comment (asraa) [reply internal/builders/docker/pkg/builder_test.go:175]: this is in a test though

File: internal/builders/docker/pkg/config.go (hunk 1 above)
Comment (asraa) [reply internal/builders/docker/pkg/config.go:89]: yes! it's read at the end of the file with safe read.

**Pull request merged**